pyccl/core.py

```diff
@@ -1148,3 +933,278 @@ def __init__(self, **kwargs):
+        growth (:obj:`dict`): a dictionary describing the linear growth of
+            matter fluctuations. It must contain three mandatory entries:
+            `'a'`: an array of monotonically ascending scale-factor
+            values. `'growth_factor'`: an array containing the values of
+            the linear growth factor :math:`D(a)` at the scale factor
+            values stored in `a`. '`growth_rate`': an array containing the
+            growth rate :math:`f(a)\\equiv d\\log D/d\\log a` at the scale
+            factor values stored in `a`.
+        pk_linear (:obj:`dict`): a dictionary containing linear power
+            spectra. It must contain the following mandatory entries:
+            `'a'`: an array of scale factor values. `'k'`: an array of
+            comoving wavenumbers in units of inverse Mpc.
+            `'delta_matter_x_delta_matter'`: a 2D array of shape
+            `(n_a, n_k)`, where `n_a` and `n_k` are the lengths of
+            `'a'` and `'k'` respectively, containing the linear matter
+            power spectrum :math:`P(k,a)`. This dictionary may also
+            contain other entries with keys of the form `'q1_x_q2'`,
+            containing other cross-power spectra between quantities
+            `'q1'` and `'q2'`.
+        pk_nonlin (:obj:`dict`): a dictionary containing non-linear
+            power spectra. It must contain the following mandatory
+            entries: `'a'`: an array of scale factor values.
+            `'k'`: an array of comoving wavenumbers in units of
+            inverse Mpc. If `use_halofit == None`, it should also
+            contain `'delta_matter_x_delta_matter'`: a 2D array of
+            shape `(n_a, n_k)`, where `n_a` and `n_k` are the lengths
+            of `'a'` and `'k'` respectively, containing the non-linear
+            matter power spectrum :math:`P(k,a)`. This dictionary may
+            also contain other entries with keys of the form `'q1_x_q2'`,
+            containing other cross-power spectra between quantities
+            `'q1'` and `'q2'`.
+        use_halofit (:obj:`bool`): if `True`, all entries in `pk_linear`
+            which do not appear in `pk_nonlin`, will be populated in the
+            latter by applying the "HALOFIT" transformation of
+            Takahashi et al. 2012 (arXiv:1208.2701) on their linear
+            versions.
+    """
+    def __init__(
+            self, Omega_c=None, Omega_b=None, h=None, n_s=None,
+            sigma8=None, A_s=None, Omega_k=0., Omega_g=None,
+            Neff=3.046, m_nu=0., m_nu_type=None, w0=-1., wa=0.,
+            T_CMB=None, background=None, growth=None,
+            pk_linear=None, pk_nonlin=None, use_halofit=False):
+        if pk_linear:
+            transfer_function = 'calculator'
+        else:
+            transfer_function = 'boltzmann_camb'
+        if pk_nonlin or use_halofit:
+            matter_power_spectrum = 'calculator'
+        else:
+            matter_power_spectrum = 'halofit'
```

Comment:
The API in this if else block is confusing to say the least. If I only pass pk_linear, then use_halofit being set or not won't matter because the base cosmology object will always apply halofit?

Reply:
See comments below